core/dataset.cpp

```diff
@@ -716,6 +721,10 @@ bool DatasetConstProxy::operator!=(const DatasetConstProxy &other) const {
   return !dataset_equals(*this, other);
 }
 
+Dimensions DatasetConstProxy::dimensions() const {
+  return m_dataset->dimensions();
```

Comment:
Forwarding dimensions from original dataset might not be the right thing to do here. We are not accounting for slices in `m_slices`.

Reply:
In fact it is wrong. Furthermore, the dimensions of `Dataset` (and its proxies) should **not** use `class Dimensions`. `Dimensions` implies an ordering, but a dataset has no well defined dimension order. This is the reason why I have not added such a method.